FirebaseSessions/Sources/ApplicationInfo.swift

```diff
@@ -33,16 +40,21 @@ protocol ApplicationInfoProtocol {
 
   /// Validated Mobile Country Code and Mobile Network Code
   var mccMNC: String { get }
+  
+  /// Development environment on which the application is running.
+  var environment: DevEnvironment { get }
 }
 
 class ApplicationInfo: ApplicationInfoProtocol {
   let appID: String
 
   private let networkInfo: NetworkInfoProtocol
+  private let envParams: [String : String]
 
-  init(appID: String, networkInfo: NetworkInfoProtocol = NetworkInfo()) {
+  init(appID: String, networkInfo: NetworkInfoProtocol = NetworkInfo(), envParams: [String : String] = ProcessInfo.processInfo.environment) {
```

Comment:
For this, you can use command line arguments as a dictionary by accessing UserDefaults (It might also match the envParams type)

If I setup a command line argument `-Hello 123`:
![Screen Shot 2022-11-09 at 12 39 43 PM](https://user-images.githubusercontent.com/555046/200901646-1eb1add6-e0a3-422d-af60-6fb65503a438.png)

Then I can access it in `UserDefaults.standard` or if you want a dictionary: `UserDefaults.standard.dictionaryRepresentation()`. The one thing is you have to leave off the `-` in the argument.

![Screen Shot 2022-11-09 at 12 40 00 PM](https://user-images.githubusercontent.com/555046/200901650-6ee4c73c-5755-4dac-aa60-e12f2290df37.png)

Reply:
UserDefaults might not be a great fit for this. UserDefaults are typically used for storing users preferences and using that for capturing log environment does not look appropriate to me. So, for now, I will go ahead use this as an environment variable (like we discussed offline).